File: python/diceroller.py
import random

print('How many dice would you like to roll?')
amount_of_dice = int(input('number> '))

print('How many sides should each die have?')
amount_of_sides = int(input('number> '))

total = 0

for i in range(amount_of_dice):
	total += random.randint(1, amount_of_sides)

print('Total: %s' %total)

File: go/diceroller.go
package main

import (
        "fmt"
        "log"
        "math/rand" // for random numbers
)

func main() {
        fmt.Println(`Dice roller

In this game, I roll as many dice as you want with as many sides as you want.`)

	// declare the variables
        var amount int
        var sides int
        var total int

        fmt.Println("How many dice would you like to roll?")
        _, err := fmt.Scan(&amount)
        if err != nil {
                log.Fatal("Invalid input.")
        }

        fmt.Println("How many sides should the dice have?")
        _, err = fmt.Scan(&sides)
        if err != nil {
                log.Fatal("Invalid input.")
        }

        for i := 0; i < amount; i++ {
                total += rand.Intn(sides) + 1 // add to total
        }

        fmt.Println("The total is", total)
}
